pkg/pipelineascode/cancel_pipelineruns.go

```diff
@@ -24,18 +24,19 @@ var cancelMergePatch = map[string]interface{}{
 }
 
 func (p *PacRun) cancelPipelineRuns(ctx context.Context, repo *v1alpha1.Repository) error {
-	if p.event.TriggerTarget != "pull_request" {
-		msg := fmt.Sprintf("not a pullRequest event, event: %v", p.event.TriggerTarget)
-		p.eventEmitter.EmitMessage(repo, zap.WarnLevel, "RepositoryEvent", msg)
-		return nil
+	labelSelector := getLabelSelector(map[string]string{
+		keys.URLRepository: formatting.CleanValueKubernetes(p.event.Repository),
+		keys.SHA:           formatting.CleanValueKubernetes(p.event.SHA),
+	})
+
+	if p.event.TriggerTarget == "pull_request" {
+		labelSelector = getLabelSelector(map[string]string{
+			keys.PullRequest: strconv.Itoa(p.event.PullRequestNumber),
+		})
 	}
 
 	prs, err := p.run.Clients.Tekton.TektonV1().PipelineRuns(repo.Namespace).List(ctx, metav1.ListOptions{
-		LabelSelector: getLabelSelector(map[string]string{
-			keys.URLRepository: formatting.CleanValueKubernetes(p.event.Repository),
-			keys.SHA:           formatting.CleanValueKubernetes(p.event.SHA),
-			keys.PullRequest:   strconv.Itoa(p.event.PullRequestNumber),
-		}),
```

Comment:
keys.PullRequest is not there anymore, is that intended? i know in case of push, it doesn't exist, but is get filled somehow later on?

Reply:
> keys.PullRequest is not there anymore 

I see its still there in the main code
https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/pipelineascode/cancel_pipelineruns.go#L37C4-L38C1

> , is that intended?

No i kept as it is like old code for pull_request

> i know in case of push, it doesn't exist, but is get filled somehow later on?

It will be 0 i think for push request

Reply:
you didn't kept as is since the pull request keys is now missing compared as before, but if we are sure of our e2e coverage for this feature (on github) then that's fine

Reply:
E2E works

> you didn't kept as is since the pull request keys is now missing compared as before,

I m not sure if i understand you question
But changes for pull_request is same like before 

I mean
**before:**
```
LabelSelector: getLabelSelector(map[string]string{
			keys.URLRepository: formatting.CleanValueKubernetes(p.event.Repository),
			keys.SHA:           formatting.CleanValueKubernetes(p.event.SHA),
			keys.PullRequest:   strconv.Itoa(p.event.PullRequestNumber),
		}),
```

after : 

```
	labelSelector := getLabelSelector(map[string]string{
		keys.URLRepository: formatting.CleanValueKubernetes(p.event.Repository),
		keys.SHA:           formatting.CleanValueKubernetes(p.event.SHA),
	})

	if p.event.TriggerTarget == "pull_request" {
		labelSelector = getLabelSelector(map[string]string{
			keys.PullRequest: strconv.Itoa(p.event.PullRequestNumber),
		})
	}
```

So for pull_request all 3 previous keys will be there

I did verify e2e for pull_request and its success